### PR TITLE
Update to version 8.2.0

### PR DESCRIPTION
**Pull request by JanWielemaker**
* Update to version 8.2.0
* Include YAML package

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.4 19E287
Xcode 11.5 11E608c 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/swi-prolog/Portfile [hunk 1]
```diff
@@ -1,63 +1,64 @@
-PortSystem  1.0
-PortGroup   cmake 1.1
+PortSystem     1.0
```
Comment (reneeotten) [inline lang/swi-prolog/Portfile:1]: please add a [modeline](https://guide.macports.org/chunked/development.creating-portfile.html) as the first line followed by an empty line:
```
# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
```

from that you'll see that the whitespace changes you made are not quite correct (they should be multiples of four). Additionally, we prefer non-functional changes in their own commit.

File: lang/swi-prolog/Portfile [hunk 2]
```diff
@@ -1,63 +1,64 @@
...
+version        8.2.0
 
-categories  lang
-license     BSD
-maintainers {uva.nl:J.Wielemaker @JanWielemaker}
-platforms   darwin
-description SWI-Prolog compiler plus extra packages (stable version)
+categories     lang
+license        BSD
+maintainers    {cwi.nl:J.Wielemaker @JanWielemaker}
```
Comment (reneeotten) [inline lang/swi-prolog/Portfile:11]: can `openmaintainer` be added here?

File: lang/swi-prolog/Portfile [hunk 3]
```diff
@@ -1,63 +1,64 @@
...
+	C/C++-interface, Multiple threads, GNU-readline interface,  \
+	coroutining, constraint programming, global variables,      \
+	very fast compiler. Including packages clib (Unix process   \
+	control, sockets, MIME), cpp (C++ interface), sgml (reading \
+	XML/SGML), sgml/RDF (reading RDF into triples), ODBC        \
+	interface and XPCE (Graphics UI toolkit, integrated editor  \
 	(Emacs-clone) and graphical debugger).
 
 homepage       http://www.swi-prolog.org/
 master_sites   http://www.swi-prolog.org/download/stable/src/
```
Comment (reneeotten) [inline lang/swi-prolog/Portfile:27]: pleases use `https` for both `homepage and` `master_sites`

File: lang/swi-prolog/Portfile [hunk 4]
```diff
@@ -1,63 +1,64 @@
...
+	port:pcre             \
+	port:readline         \
+	port:Xft2             \
+	port:xorg-libXinerama \
+	port:xorg-libXt       \
+	port:xpm              \
+	port:libyaml          \
+	port:zlib
+
+use_parallel_build    yes
```
Comment (reneeotten) [inline lang/swi-prolog/Portfile:58]: this is the default and can thus be removed

File: lang/swi-prolog/Portfile [hunk 5]
```diff
@@ -71,12 +72,11 @@ configure.pre_args  \
 	-DCMAKE_INCLUDE_PATH=${prefix}/include      \
 	-DCMAKE_LIBRARY_PATH=/usr/lib:${prefix}/lib
 
-pre-build {
-	system "cd ${worksrcpath} && ./scripts/pgo-compile.sh"
-}
+test.run no
+test.cmd ctest --output-on-failure
```
Comment (reneeotten) [inline lang/swi-prolog/Portfile:76]: if you do not want to run tests (default), you don't need to set these two `test.*` fields.

**Comment by JanWielemaker** [comment]
Thanks for the comments.  I'll make a new PR as you want squashed PRs.   The test is a bit unfortunate.  Due to some oddity with the macport build environment regarding /tmp, one of the tests fails.   I cannot update the source and going through the whole thing of adding patches seems overkill.

This is resolved in the 8.3.0 devel series (where I also dealt with the above requests) and will be ok for 8.2.1 (stable).

**Pull request closed without merging**